### PR TITLE
Various bug fixes

### PR DESCRIPTION
**Pull request by kamorel**
# Description

[f32b228](https://github.com/bcgov/bcbox/commit/f32b228b094edc5925f325de2c64f845e1fb17fa) - Show bucket permissions icon to MANAGE users only
[5b2c91d](https://github.com/bcgov/bcbox/commit/5b2c91d7a72f3bb26700af97147b871a2c5b4bb2) - Implement breadcrumb navigation

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/layout/Navbar.vue [hunk 1]
```diff
@@ -32,6 +33,16 @@ const { getKeycloak } = storeToRefs(useAuthStore());
         </ol>
       </template>
     </Toolbar>
+
+    <!-- Breadcrumb
```
Comment (dc-bcgov) [inline frontend/src/components/layout/Navbar.vue:37]: Wondering if we need to keep the commented code?

File: frontend/src/components/object/ObjectList.vue [hunk 2]
```diff
@@ -54,62 +54,76 @@ const listObjects = async () => {
   }
 };
 
+// const updateBreadcrumb = async () => {
```
Comment (dc-bcgov) [inline frontend/src/components/object/ObjectList.vue:57]: Same here - wondering if we need to keep the commented code?

File: frontend/src/components/object/ObjectList.vue [hunk 3]
```diff
@@ -54,62 +54,76 @@ const listObjects = async () => {
...
+
 // Download
 const multiSelectedObjectIds = computed(() => {
   return multiSelectedObjects.value.map((o) => o.id);
 });
 
 // Get the user's list of objects
 onMounted(() => {
+  // Removed for now
+  // updateBreadcrumb();
```
Comment (dc-bcgov) [inline frontend/src/components/object/ObjectList.vue:74]: Same here - wondering if we need to keep the commented code?

File: frontend/src/router/index.ts [hunk 4]
```diff
@@ -67,6 +67,10 @@ const router = createRouter({
 });
 
 router.beforeEach((to) => {
+  // Removed for now
+  // const navStore = useNavStore();
```
Comment (dc-bcgov) [inline frontend/src/router/index.ts:71]: Same here - wondering if we need to keep the commented code?

**Review by dc-bcgov — APPROVED** [review]

File: frontend/src/components/layout/Navbar.vue (hunk 1 above)
Comment (jujaga) [reply frontend/src/components/layout/Navbar.vue:37]: For this one, we want to revisit breadcrumbs at a later date, so this should be preserved.

File: frontend/src/components/object/ObjectList.vue (hunk 2 above)
Comment (jujaga) [reply frontend/src/components/object/ObjectList.vue:57]: For this one, we want to revisit breadcrumbs at a later date, so this should be preserved.

File: frontend/src/components/object/ObjectList.vue (hunk 3 above)
Comment (jujaga) [reply frontend/src/components/object/ObjectList.vue:74]: For this one, we want to revisit breadcrumbs at a later date, so this should be preserved.

File: frontend/src/components/object/ObjectList.vue [hunk 5]
```diff
@@ -54,62 +54,76 @@ const listObjects = async () => {
...
   listObjects();
+
```
Comment (jujaga) [inline frontend/src/components/object/ObjectList.vue:76]: Extra whitespace line

File: frontend/src/router/index.ts (hunk 4 above)
Comment (jujaga) [reply frontend/src/router/index.ts:71]: For this one, we want to revisit breadcrumbs at a later date, so this should be preserved.

File: frontend/src/views/ListObjectsView.vue [hunk 6]
```diff
@@ -1,12 +1,53 @@
 <script setup lang="ts">
+import { ref, Ref } from 'vue';
+import { useRoute } from 'vue-router';
+// PrimeVue
+import { useToast } from 'primevue/usetoast';
+// State
+import { useBucketStore} from '@/store';
```
Comment (jujaga) [inline frontend/src/views/ListObjectsView.vue:7]: Missing space before }

**Review by jujaga — APPROVED** [review]

File: frontend/src/components/bucket/BucketsTable.vue [hunk 7]
```diff
@@ -27,6 +28,10 @@ const showPermissions = async (bucketId: string, bucketName: string) => {
   permissionsBucketId.value = bucketId;
   permissionBucketName.value = bucketName;
 };
+
+const displayPermissionsIcon = (bucket: Bucket) => {
+  return bucket.userPermissions?.find( (x: Permission) => x.permCode === Permissions.MANAGE);
+};
```
Comment (TimCsaky) [inline frontend/src/components/bucket/BucketsTable.vue:34]: could consider passing permCode as a param.
later we may want to do `displayIcon(<bucket uuid>, DELETE)`

File: frontend/src/components/object/ObjectList.vue [hunk 8]
```diff
@@ -54,61 +54,74 @@ const listObjects = async () => {
...
+      <div class="flex-grow-1">
+        <ObjectTable
+          :display-info="displayInfo"
+          @show-info="showInfo"
+        />
+      </div>
+      <div
+        v-if="displayInfo"
+        class="flex-shrink-0 ml-3"
+        style="max-width: 33%; min-width: 33%"
```
Comment (TimCsaky) [inline frontend/src/components/object/ObjectList.vue:118]: this seems like a good flexible alternative to just doing `width: 33%`

File: frontend/src/store/bucketStore.ts [hunk 9]
```diff
@@ -105,12 +111,12 @@ export const useBucketStore = defineStore('bucket', () => {
...
       loading.value = true;
-  
+
       if (currentUser.value) {
-        const bucketPerms = 
+        const bucketPerms =
           (await permissionService.bucketSearchPermissions({ bucketId, userId: currentUser.value.userId })).data;
-        if(bucketPerms && bucketPerms[0] && bucketPerms[0].permissions) {
-          selectedBucketPermissionsForUser.value =bucketPerms[0].permissions;
+        if (bucketPerms && bucketPerms[0] && bucketPerms[0].permissions) {
```
Comment (TimCsaky) [inline frontend/src/store/bucketStore.ts:118]: the permission search in coms wll return an empty array if no perms were found.
maybe you can skip line 115-118 and do:
`selectedBucketPermissionsForUser.value = await permissionService.bucketSearchPermissions({ bucketId, userId: currentUser.value.userId })).data.[0].permissions`

**Review by TimCsaky — APPROVED** [review]
minor comments.. happy for it to be merged

**Pull request merged**